docs/snackbar.js

```diff
@@ -0,0 +1,67 @@
+export {showSnackbar};
+
+const ANIMATION_DURATION_MS = 500;
+const SNACKBAR_DURATION_MS = 3000;
+
+// Track number of messages shown so we don't accidentally hide a later message.
+let showNumber = 0;
+
+let snackbarOriginalClassName;
+let snackbarOriginalStyle;
+let iconOriginalClassName;
+let iconOriginalStyle;
+
+/**
+ * Shows a snackbar notification.
+ *
+ * @param {string} msg the message to be shown
+ * @param {function} snackbarStyler applies styles to the snackbar div
+ * @param {function} iconStyler applies styles to the icon div
+ * @param {number} duration Number of milliseconds to show the message. If
+ *     non-positive, message stays up until this method is called again.
+ */
+function showSnackbar(
+    msg, snackbarStyler, iconStyler, duration = SNACKBAR_DURATION_MS) {
+  const snackbar = document.getElementById('snackbar');
+  if (snackbarStyler) snackbarStyler(snackbar);
```

Comment:
I think you're only setting the snackbar element's style. Might be easier to just pass the styles in as a dict and the classnames in as a list and then apply it yourself in here. That also guarantees that the lambda didn't do anything else to the element, since you're only resetting the classes and style. Same goes for icon. Maybe we should only allow setting class names, and then the toast and the error styles can be classes?

Reply:
Fair enough